pkg/i2gw/providers/ingressnginx/server_alias_test.go

```diff
@@ -0,0 +1,330 @@
+				Gateways: map[types.NamespacedName]intermediate.GatewayContext{
+					{Name: "test-gateway", Namespace: "default"}: {
+						Gateway: gatewayv1.Gateway{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-gateway",
+								Namespace: "default",
+							},
+							Spec: gatewayv1.GatewaySpec{
+								Listeners: []gatewayv1.Listener{
+									{
+										Name:     "http",
+										Protocol: gatewayv1.HTTPProtocolType,
+										Port:     80,
+										Hostname: ptrTo(gatewayv1.Hostname("example.com")),
+									},
+									{
+										Name:     "api-example-com",
+										Protocol: gatewayv1.HTTPProtocolType,
+										Port:     80,
+										Hostname: ptrTo(gatewayv1.Hostname("api.example.com")),
+									},
+									{
+										Name:     "cdn-example-com",
+										Protocol: gatewayv1.HTTPProtocolType,
+										Port:     80,
+										Hostname: ptrTo(gatewayv1.Hostname("cdn.example.com")),
+									},
+								},
+							},
+						},
+					},
+				},
+				HTTPRoutes: map[types.NamespacedName]intermediate.HTTPRouteContext{
+					{Name: "test-route", Namespace: "default"}: {
+						HTTPRoute: gatewayv1.HTTPRoute{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-route",
+								Namespace: "default",
+							},
+							Spec: gatewayv1.HTTPRouteSpec{
+								Hostnames: []gatewayv1.Hostname{
+									"example.com",
+									"api.example.com",
+									"cdn.example.com",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "ingress without server alias annotation",
+			ingresses: []networkingv1.Ingress{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-ingress",
+						Namespace: "default",
+					},
+					Spec: networkingv1.IngressSpec{
+						Rules: []networkingv1.IngressRule{
+							{
+								Host: "example.com",
+							},
+						},
+					},
+				},
+			},
+			initialIR: intermediate.IR{
+				Gateways:   map[types.NamespacedName]intermediate.GatewayContext{},
+				HTTPRoutes: map[types.NamespacedName]intermediate.HTTPRouteContext{},
+			},
+			expectedIR: intermediate.IR{
+				Gateways:   map[types.NamespacedName]intermediate.GatewayContext{},
+				HTTPRoutes: map[types.NamespacedName]intermediate.HTTPRouteContext{},
+			},
+			expectedErrors: 0,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ir := tc.initialIR
+			errs := serverAliasFeature(tc.ingresses, nil, &ir)
+
+			if len(errs) != tc.expectedErrors {
+				t.Fatalf("Expected %d errors, got %d: %v", tc.expectedErrors, len(errs), errs)
+			}
+
+			// Compare gateways
+			if diff := cmp.Diff(tc.expectedIR.Gateways, ir.Gateways); diff != "" {
+				t.Fatalf("Gateways mismatch (-want +got):\n%s", diff)
+			}
+
+			// Compare HTTPRoutes
+			if diff := cmp.Diff(tc.expectedIR.HTTPRoutes, ir.HTTPRoutes); diff != "" {
+				t.Fatalf("HTTPRoutes mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+func TestGetIngressClass(t *testing.T) {
```

Comment:
why is this in server_alias_test.go? was it supposed to be on #231 ?

Reply:
It's to test https://github.com/kubernetes-sigs/ingress2gateway/pull/227/files#diff-390c4deeef1d9cf7dca46e3444ce4565ca9057fbf900ce1c4740cfdfcffa5ab9R137-R146